shell/common/rasterizer.cc

```diff
@@ -802,28 +810,143 @@ static sk_sp<SkData> ScreenshotLayerTreeAsPicture(
+    flutter::LayerTree* tree,
     GrDirectContext* surface_context,
+    const std::shared_ptr<impeller::AiksContext>& aiks_context) {
+  // There is no root surface transformation for the screenshot layer. Reset
+  // the matrix to identity.
+  SkMatrix root_surface_transformation;
+  root_surface_transformation.reset();
+
+  auto frame = compositor_context.AcquireFrame(
+      surface_context,              // skia context
+      canvas,                       // canvas
+      nullptr,                      // view embedder
+      root_surface_transformation,  // root surface transformation
+      false,                        // instrumentation enabled
+      true,                         // render buffer readback supported
+      nullptr,                      // thread merger
+      aiks_context.get()            // aiks context
+  );
+  canvas->Clear(DlColor::kTransparent());
+  frame->Raster(*tree, true, nullptr);
+  canvas->Flush();
+}
+
+#if IMPELLER_SUPPORTS_RENDERING
+Rasterizer::ScreenshotFormat ToScreenshotFormat(impeller::PixelFormat format) {
+  switch (format) {
+    case impeller::PixelFormat::kUnknown:
+    case impeller::PixelFormat::kA8UNormInt:
+    case impeller::PixelFormat::kR8UNormInt:
+    case impeller::PixelFormat::kR8G8UNormInt:
+    case impeller::PixelFormat::kR8G8B8A8UNormIntSRGB:
+    case impeller::PixelFormat::kB8G8R8A8UNormIntSRGB:
+    case impeller::PixelFormat::kB10G10R10XRSRGB:
+    case impeller::PixelFormat::kS8UInt:
+    case impeller::PixelFormat::kD24UnormS8Uint:
+    case impeller::PixelFormat::kD32FloatS8UInt:
+    case impeller::PixelFormat::kR32G32B32A32Float:
+    case impeller::PixelFormat::kB10G10R10XR:
+    case impeller::PixelFormat::kB10G10R10A10XR:
+      FML_DCHECK(false);
+      return Rasterizer::ScreenshotFormat::kUnknown;
+    case impeller::PixelFormat::kR8G8B8A8UNormInt:
+      return Rasterizer::ScreenshotFormat::kR8G8B8A8UNormInt;
+    case impeller::PixelFormat::kB8G8R8A8UNormInt:
+      return Rasterizer::ScreenshotFormat::kB8G8R8A8UNormInt;
+    case impeller::PixelFormat::kR16G16B16A16Float:
+      return Rasterizer::ScreenshotFormat::kR16G16B16A16Float;
+  }
+}
+
+static std::pair<sk_sp<SkData>, Rasterizer::ScreenshotFormat>
+ScreenshotLayerTreeAsImageImpeller(
+    const std::shared_ptr<impeller::AiksContext>& aiks_context,
+    flutter::LayerTree* tree,
+    flutter::CompositorContext& compositor_context,
+    bool compressed) {
+  if (compressed) {
+    FML_LOG(ERROR) << "Compressed screenshots not supported for Impeller";
+    return {nullptr, Rasterizer::ScreenshotFormat::kUnknown};
+  }
+
+  DisplayListBuilder builder(SkRect::MakeSize(
+      SkSize::Make(tree->frame_size().fWidth, tree->frame_size().fHeight)));
+
+  RenderFrameForScreenshot(compositor_context, &builder, tree, nullptr,
+                           aiks_context);
+
+  impeller::DlDispatcher dispatcher;
+  builder.Build()->Dispatch(dispatcher);
+  const auto& picture = dispatcher.EndRecordingAsPicture();
```

Comment:
https://google.github.io/styleguide/cppguide.html#Type_deduction

Reply:
I think this one is pretty clear.

Reply:
It's not if it is clear to you, "Use type deduction only if it makes the code clearer to readers who aren't familiar with the project, or if it makes the code safer. Do not use it merely to avoid the inconvenience of writing an explicit type."

This isn't my personal style, I'm just reporting based on the style guide we've agreed to follow =)